### ISSUE TITLE
Question on num_boxes when calculating loss

### ISSUE THREAD

**Issue by HYUNJS** [facebookresearch/detr#267]
Hello, guys :)

I have a question regarding the code used in loss computation. More specifically, the code related to calculating `num_boxes` in here https://github.com/facebookresearch/detr/blob/master/models/detr.py#L232. 

Why do we divide the number of bboxes by the number of gpus?

`num_boxes = torch.clamp(num_boxes / get_world_size(), min=1).item()`

**Comment by alcinos** [comment]
Hi,
Relevant quote from the paper:

> All losses are normalized by the number of objects inside the batch. Extra care must betaken  for  distributed  training:  since  each  GPU  receives  a  sub-batch,  it  is  notsufficient  to  normalize  by  the  number  of  objects  in  the  local  batch,  since  ingeneral the sub-batches are not balanced across GPUs. Instead, it is importantto normalize by the total number of objects in all sub-batches

The reason is that if a batch on one gpu gets a lot of objects, while another batch gets fewer, if you normalize by number of gpus the loss per object will not be balanced (and favor the batch with few objects, which are typically easy)

I believe I have answered your question, and as such I'm closing this. Feel free to reach out if you have further questions.
